### PR TITLE
fix(git): Use absolute path instead

### PR DESCRIPTION
**Pull request by StealthyCoder**
Installing `git` on Windows will most likely place it in the `Program Files` directory.

Following our own [docs](https://docs.foundries.io/latest/getting-started/install-fioctl/index.html#configuring-git) on Windows will run the command as Administrator and place the resulting credential helper in `C:\Program Files\Git\bin\`. In order to make it work as seamless as possible on Windows we use the [absolute path configuration](https://git-scm.com/docs/gitcredentials#_custom_helpers) for the credential helper in the `.gitconfig` file managing the global configuration for git. 

On Windows the absolute path has to be converted into a different format.

1. All `\` have to become `/`
2. All spaces have to be 'escaped' making a ` ` into `\\ `

This works around symlinks, permission errors and other edge cases on Windows.

### CODE REVIEW

File: subcommands/git/cmd.go
Comment (vkhoroz) [inline subcommands/git/cmd.go:86]: hmm... this is interesting...

So, as I understand the `git` utility under Windows sometimes (or always) runs in a Linux-like shell; otherwise, the conversion to slashes would not work (Windows uses backslashes).

For the first replace (backslash-to-slash), Golang has a [filepath.ToSlash function](https://pkg.go.dev/path/filepath#ToSlash).

Then, a problem with spaces is IMHO not related to Windows, it seems generic enough to be handled outside this `if` clause.
I suspect this same problem could happen in Linux had the `git` utility contain a space in its path.
Thus, if I were you I would try putting the `helperName` into double quotes at lines 94 and 98 below.

File: subcommands/git/cmd.go
Comment (vkhoroz) [inline subcommands/git/cmd.go:84]: Once done, try to truncate this comment to at most 2 lines.

File: subcommands/git/cmd.go
Comment (StealthyCoder) [reply subcommands/git/cmd.go:82]: Tried that, git on Windows will then treat it as a string aka name, and will prepend it with git-credential again :joy:

Comment (StealthyCoder) [reply subcommands/git/cmd.go:82]: Did not know the ToSlash function. Will utilize that for the first part.

File: subcommands/git/cmd.go
Comment (doanac) [reply subcommands/git/cmd.go:80]: nitpick - but this is an unusual way to do comments in golang (the two asterisk lines). https://tip.golang.org/doc/comment

File: subcommands/git/cmd.go
Comment (doanac) [reply subcommands/git/cmd.go:82]: Seems like the fix should be in the code that sets `helperPath` and not here. That way the user sees the correct default value in the usage message. Also - have you looked at `filepath.Abs`? I'd think that would work correctly on windows.

Comment (StealthyCoder) [reply subcommands/git/cmd.go:82]: I will look into `filepath.Abs` as well. To be clear, this format I need to put in there is specific for how `git` on Windows needs it to be in order to run. On a related note, we are [not the only ones struggling with spaces on Windows and commands and formatting](https://github.com/golang/go/issues/17149).

Comment (StealthyCoder) [reply subcommands/git/cmd.go:82]: `filepath.Abs` sadly does not give that specific format. :slightly_smiling_face:

File: subcommands/git/cmd.go
Comment (StealthyCoder) [reply subcommands/git/cmd.go]: Fixed this.

File: subcommands/git/cmd.go
Comment (doanac) [reply subcommands/git/cmd.go:80]: k. lgtm then. i'll let volodya approve

Comment (vkhoroz) [reply subcommands/git/cmd.go:80]: Did you check if the Git on Linux processes spaces inside the path correctly?
I suspect it would have the same problem, in which case we need space escaping to be common.

Comment (StealthyCoder) [reply subcommands/git/cmd.go]: I never tried to do it in Linux yet, with absolute paths. So I will try and see what I can do there as well. I also realized I put the logic in the wrong place. The `helperName` should be changed, not `dst`. So I fixed that now. Tested it on Windows and it works correctly now. 

Let me see what happens on Linux when given a file path with spaces and absolute path.

Comment (StealthyCoder) [reply subcommands/git/cmd.go:82]: Added some logic that detects the usage of `creds-path`, then always use the full path. Most likely scenario on Linux is somebody does not have something in their `PATH` variable we can write to, like a fresh install of any distro and then they give the `--creds-path` and on that change we use the full path in the git config as well.

Comment (vkhoroz) [reply subcommands/git/cmd.go:82]: I'm not sure I understand the `creds-path` logic; but I guess you know what you do here.

As long as it works I'm fine.

**Review by vkhoroz — APPROVED** [review]

File: subcommands/git/cmd.go
Comment (doanac) [inline subcommands/git/cmd.go:85]: I'm now confused by this whole PR. I think the normal case (not windows i guess) is to:
 * find a directory in `$PATH` we can write to
 * link file fioctl -> git-credential-fio
 * run `git config ... ` to set the helper

The whole point of finding something in $PATH is so we don't need absolute paths?

It almost sounds like on windows - we aren't finding something in the user's actual path?

If we do need absolute paths, rather than trying to "encode" (nitpick - I think we are *escaping*) - we probably need to put quotes around what we present to git-config. That way what winds up in git config looks like:
```
helper = "c:\\program files\blah.exe"
```
I suspect the quotations would address things beyond spaces we haven't considered.

That said - seems like the bug is that we can't find PATH correctly on windows

Comment (StealthyCoder) [reply subcommands/git/cmd.go]: I agree, the problem partly stems from us requiring the user to run as `root`/`Administrator` and the `PATHS` being different for them on Windows. I will briefly look into a way if you can _easily_ get the original calling user on Windows. If that is the case we could either add in logic to add things to the PATH for that user, or place the symlink into one of the places in their PATH locations. Then we can drop this special case for Windows again, apart from the naming having an extra `.exe`. :+1:

**Review by vkhoroz — APPROVED** [review]
LGTM

I'm fine to merge this as long as @doanac is fine with it.

**Review by doanac — APPROVED** [review]

**Pull request merged**
